service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/dataset/CopyBigQueryDatasetDefinitionStep.java

```diff
@@ -117,8 +117,8 @@ public StepResult doStep(FlightContext flightContext)
             .sourceWorkspaceId(sourceDataset.getWorkspaceId())
             .sourceResourceId(sourceDataset.getResourceId());
     workingMap.put(ControlledResourceKeys.CLONE_DEFINITION_RESULT, apiResult);
-    if (CloningInstructions.COPY_DEFINITION.equals(effectiveCloningInstructions)) {
-      // we're done
+    if (CloningInstructions.COPY_DEFINITION.equals(effectiveCloningInstructions)
+        || CloningInstructions.COPY_RESOURCE.equals(effectiveCloningInstructions)) {
```

Comment:
nit: maybe a comment about why you can set the response early for the COPY_RESOURCE case?